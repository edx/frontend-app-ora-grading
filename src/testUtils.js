// eslint-disable-next-line import/prefer-default-export
export const formatMessage = (msg, values) => {
  let message = msg.defaultMessage;
  if (values === undefined) { return message; }
  Object.keys(values).forEach(key => {
    // eslint-disable-next-line
    message = message.replace(`{${key}}`, values[key]);
  });
  return message;
};
