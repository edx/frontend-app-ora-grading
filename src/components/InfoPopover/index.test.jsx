import React from 'react';
import { shallow } from 'enzyme';

import { formatMessage } from 'testUtils';
import { InfoPopover } from '.';

describe('Info Popover Component', () => {
  const child = <div>Children component</div>;
  const onClick = jest.fn().mockName('this.props.onClick');
  let el;
  beforeEach(() => {
    el = shallow(<InfoPopover onClick={onClick} intl={{ formatMessage }}>{child}</InfoPopover>);
  });
  test('snapshot', () => {
    expect(el).toMatchSnapshot();
  });
  describe('Component', () => {
    test('Test component render', () => {
      expect(el.length).toEqual(1);
      expect(el.find('.esg-help-icon').length).toEqual(1);
    });
  });
});
