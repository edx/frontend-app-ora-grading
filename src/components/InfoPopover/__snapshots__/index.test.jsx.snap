// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info Popover Component snapshot 1`] = `undefined`;
